Droidcon-Boston/app/src/main/java/com/mentalmachines/droidcon_boston/firebase/DbFirebaseMessagingService.java

```diff
@@ -0,0 +1,41 @@
+package com.mentalmachines.droidcon_boston.firebase;
+
+import android.util.Log;
+import com.google.firebase.messaging.FirebaseMessagingService;
+import com.google.firebase.messaging.RemoteMessage;
+import com.mentalmachines.droidcon_boston.utils.NotificationUtils;
+import java.util.Map;
+
+public class DbFirebaseMessagingService extends FirebaseMessagingService {
+
+    private static final String TAG = "FCM";
+
+    @Override
+    public void onMessageReceived(RemoteMessage remoteMessage) {
+
+        if (remoteMessage != null) {
+            Map<String, String> payloadMap = remoteMessage.getData();
+            String bodyStr = "NA";
+
+            // Check if message contains a data payload.
+            if (payloadMap.size() > 0) {
+                Log.d(TAG, "Payload: ");
+                for (String key : payloadMap.keySet()) {
+                    Log.d(TAG, "Key: " + key + ", Value: " + payloadMap.get(key));
```

Comment:
wrap Log.d in BuildConfig.DEBUG so it gets removed from release builds.
Or use Timber...it's a much better logging system.

Reply:
Proguard can take out Log.d. Let's skip Timber and use proguard for this point in a later pull request.

Reply:
I added it in the last commit I pushed.